spring-boot-project/spring-boot-docs/src/docs/antora/modules/reference/pages/testing/testcontainers.adoc

```diff
@@ -5,14 +5,69 @@ The https://www.testcontainers.org/[Testcontainers] library provides a way to ma
+The extension is activated by applying the
+
 Testcontainers can be used in a Spring Boot test as follows:
 
 include-code::vanilla/MyIntegrationTests[]
 
-This will start up a docker container running Neo4j (if Docker is running locally) before any of the tests are run.
+This will start up a Docker container running Neo4j (if Docker is running locally) before any of the tests are run.
 In most cases, you will need to configure the application to connect to the service running in the container.
 
+In this case the lifecycle of the container instance is managed by Testcontainers framework, as described in official documentation.
+
+== Using via Spring managed beans
+
+The containers provided by Testcontainers framework can be managed by Spring Boot as beans.
+This method is often used in combination with javadoc:org.springframework.boot.testcontainers.service.connection.ServiceConnection[format=annotation].
+
+To use Testcontainer contains as Spring beans we need to create a configuration class declaring the container as bean:
+
+include-code::beandeclaration/BeanDeclarationConfig[]
+
+then we can start the container by importing the configuration class in the test class:
+
+include-code::beandeclaration/SpringTest[]
+
+
+== Using via importing container declaration classes
+
+A common pattern with Testcontainers framework is to declare the Container instances as static fields in an interface
+For example the following interface `MyInterface` declares two containers, one named `mongo` of type MongoDB and another named `neo` of type Neo4j:
+
+include-code::importcontainers/MyInterface[]
+
+When you have containers declared in this way, then you can have these containers managed by Spring Boot as beans.
+All that is needed to do that is adding javadoc:org.springframework.boot.testcontainers.context.ImportTestcontainers[format=annotation] to your configuration class as in:
+
+include-code::importcontainers/MyConfiguration[]
+
+== Lifecycle of managed containers
+
+If you have used the annotations and extensions provided by Testcontainers framework, then the lifecycle of container instances is managed by the Testcontainers framework.
+Please refer to the official documentation for the information about lifecycle of the containers, when managed by the Testcontainers framework.
+
+When the containers are managed by Spring as beans, then the lifecycle is clearly defined by Spring.
+The container beans are created and started before the beans of other types are created.
+This process ensures that any beans, which rely on functionality provided by the containers, can use those functionalities.
+
+The test containers can be started multiple times.
+Like any other beans the test containers are created and started once per application context managed by the TestContext Framework.
+For details about how TestContext framework manages the underlying application contexts and beans therein, please refer to the official Spring documentation.
```

Comment:
should add link?

Reply:
See above.

Reply:
I assume that here you are refering to link to `TestContext Framework` docs.

Reply:
Added as Antora attribute and linked